src/codegen/mod.rs

```diff
@@ -3674,19 +3674,19 @@ fn objc_method_codegen(
                 .to_owned(),
         );
         quote! {
-            msg_send!(class!(#class_name), #methods_and_args)
+            unsafe { msg_send!(class!(#class_name), #methods_and_args) }
         }
     } else {
         quote! {
-            msg_send!(self, #methods_and_args)
+            unsafe { msg_send!(self, #methods_and_args) }
         }
     };
 
     let method_name =
         ctx.rust_ident(format!("{}{}", prefix, method.rust_name()));
 
     quote! {
-        unsafe fn #method_name #sig where <Self as std::ops::Deref>::Target: objc::Message + Sized {
+        fn #method_name #sig where <Self as std::ops::Deref>::Target: objc::Message + Sized {
```

Comment:
I think generating these as safe methods will allow for unsound safe code. For example, if you've generated an interface from the `NSString` header:
```
let c_str = 0x12345 as *const c_char;
let str = NSString::stringWithUTF8String_(c_str);
```

This will invoke undefined behavior for a user who has only written safe code.

Reply:
Ah. Good call. Thanks!